lib/matplotlib/tri/_triinterpolate.py

```diff
@@ -442,8 +441,9 @@ def _interpolate_single_key(self, return_key, tri_index, x, y):
                 alpha, J, ecc, dof)
             if return_key == 'dzdx':
                 return dzdx[:, 0, 0]
-            else:
+            elif return_key == 'dzdy':
                 return dzdx[:, 1, 0]
+        _api.check_in_list(['z', 'dzdx', 'dzdy'], return_key=return_key)
```

Comment:
There is something wrong either in the original, or in the change.  Maybe the list should not include 'z'?  The original gives the same result with 'z' as with 'dzdy'.  Your modification will return None with 'z'.  Attn: @ianthomas23

Reply:
I reverted this change.